youtube_dl/extractor/xnxx.py

```diff
@@ -6,17 +6,26 @@
 
 
 class XNXXIE(InfoExtractor):
-    _VALID_URL = r'^https?://(?:video|www)\.xnxx\.com/video(?P<id>[0-9]+)/(.*)'
-    _TEST = {
-        'url': 'http://video.xnxx.com/video1135332/lida_naked_funny_actress_5_',
-        'md5': '0831677e2b4761795f68d417e0b7b445',
+    _VALID_URL = r'^https?://(?:video|www)\.xnxx\.com/video-?(?P<id>[0-9a-z]+)/(.*)'
+    _TESTS = [{
+        'url': 'http://www.xnxx.com/video-43s6xfe/javso.com_-_beautiful_body_asian_softcore_idol',
+        'md5': '6675d4424033b7f27b34dc103b3ad7ae',
         'info_dict': {
+            'id': '43s6xfe',
+            'ext': 'flv',
+            'title': 'javso.com - Beautiful Body Asian Softcore Idol',
+            'age_limit': 18,
+        },
```

Comment:
I'm aware of that. Old format URLs are redirected to the new one. Testing both will only introduce unnecessary network I/O. Make it `only_matching` if you don't want to delete.